### PR TITLE
lightningd: fix two crashes in plugin controller

### PR DESCRIPTION
**Pull request by rustyrussell**
Several corner cases I hit in practice.

### CODE REVIEW

File: tests/plugins/slow_init.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from lightning import Plugin
+import time
+
+plugin = Plugin()
+
+
+@plugin.init()
+def init(options, configuration, plugin):
+    plugin.log("slow_init.py initializing")
+    time.sleep(1)
```
Comment (ZmnSCPxj) [inline tests/plugins/slow_init.py]: Would the test be reliable with just a 1-second sleep here?

**Comment by darosior** [comment]
ACK 63bc5f9124d80a1802b4222bb4a8c9f04d263ac9

File: tests/plugins/slow_init.py (hunk 1 above)
Comment (cdecker) [reply tests/plugins/slow_init.py]: Should be ok, maybe `valgrind` might be a bit too slow on CI, but 1 second is usually plenty in these cases.
